experimental/emulated_vault/opt/emulated-vault/vault-server

```diff
@@ -0,0 +1,249 @@
+#! /usr/bin/python
+
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+#
+
+import json
+import logging
+import sys
+import os
+import inspect
+from BaseHTTPServer import HTTPServer
+from BaseHTTPServer import BaseHTTPRequestHandler
+import ssl
+import urlparse, json
+import re, fnmatch
+
+import adaptor.fs
+
+class RequestHandler(BaseHTTPRequestHandler):
+
+    def do_GET(self):
+        try:
+            self._do_GET()
+        except:
+            logger.exception("do_GET exception")
+
+    do_HEAD = do_GET
+
+    def do_POST(self):
+        try:
+            self._do_POST()
+        except:
+            logger.exception("do_POST exception")
+
+    do_PUT = do_POST
+
+    def do_DELETE(self):
+        try:
+            self._do_DELETE()
+        except:
+            logger.exception("do_DELETE exception")
+
+    def _do_GET(self):
+        logger.debug("GET %s", self.path)
+        parsed_path = urlparse.urlparse(self.path)
+
+        if parsed_path.path == "/ping":
+            logger.info("Ping")
+            success, value = adaptor.ping()
+            if not success:
+                self.send_response(503)
+                self.send_header('Content-type', 'text/html')
+                self.end_headers()
+                self.wfile.write("Failure")
+                return
+
+            self.send_response(200)
+            self.send_header('Content-type', 'text/html')
+            self.end_headers()
+            self.wfile.write(value)
+            return
+
+        if parsed_path.path == "/search/query/sslkeys":
+            logger.info("Search SSL: path=%s query=%s", parsed_path.path, parsed_path.query)
+            filters = {}
+            keyFilters = {}
+            cdnFind = re.search(".*q=cdn:([^&]*).*", parsed_path.query)
```

Comment:
instead of using regular expressions to search through the query string, consider using [`urllib.parse.parse_qs`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qs) to parse out a list of values for a given query parameter

Reply:
I prefer not to mess with the logic. It was written a while ago and we do not have an automated test for it. Lets fix if it is taken to production.